### PR TITLE
feat: add wdyr

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds [`why-did-you-render`](https://github.com/welldone-software/why-did-you-render) for development, so we can track rerender madness 

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- relates to https://github.com/shapeshift/lib/issues/943

## Risk

This could not detect `development` NODE_ENV properly and run on prod, worst case scenario we don't actually commit `ComponentName.whyDidYouRender` property in prod so we're safu

### Engineering

- Add the following after any component declaration

```typescript
ComponentName.whyDidYouRender = {
  logOnDifferentValues: true,
}
```
- Development server logs re-renders
- Served `yarn build` doesn't log re-renders

### Operations

N/A

## Screenshots (if applicable)

https://user-images.githubusercontent.com/17035424/200682473-180d41e7-fc88-4fbc-a7b3-cd8f96cb8376.mov

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2319** <a href="https://app.graphite.dev/github/pr/shapeshift/web/2319" target="_blank">Graphite</a> 
    * **PR #3252** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3252" target="_blank">Graphite</a>  👈
    * **PR #3248** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3248" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3252?utm_source=stack-comment).

**Review by 0xdef1cafe — APPROVED** [review]
absolute g

**Review by 0xApotheosis — DISMISSED** [review]
Can we add this to the README somewhere?

Screenshot 2022-11-09 at 9 01 33 am

**Comment by gomesalexandre** [comment]
> 

Ab-solutely: https://github.com/shapeshift/web/pull/3252/commits/47c5b900e32eb03b55f713be26d3fd3759f45d29

This will also pave the way for a selectors stability README, including `reselect-tools`

File: docs/rerenders.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+```typescript
+ComponentName.whyDidYouRender = true
+```
+
+This will detect re-renders despite values being equal.
+
+For advanced usage, see [why-did-you-render](https://github.com/welldone-software/why-did-you-render) docs.
+
+
```
Comment (gomesalexandre) [inline docs/rerenders.md:15]:
```suggestion
```

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**